process/smartContract/vmInput.go

```diff
@@ -100,6 +109,10 @@ func (sc *scProcessor) createVMCallInput(
 	vmCallInput.VMInput.Arguments = finalArguments
 	vmCallInput.GasProvided = vmCallInput.GasProvided - gasLocked
 
+	if vmCallInput.GasProvided+vmCallInput.GasLocked > tx.GetGasLimit() {
```

Comment:
can't we use here the safe add? What happens if the sum overflows the uint64 max value?

Reply:
done